deltacat/compute/compactor/compaction_session.py

```diff
@@ -68,9 +69,10 @@ def compact_partition(
         min_hash_bucket_chunk_size: int = 0,
         compacted_file_content_type: ContentType = ContentType.PARQUET,
         delete_prev_primary_key_index: bool = False,
-        read_round_completion: bool = False,
         pg_config: Optional[PlacementGroupConfig] = None,
         schema_on_read: Optional[pa.schema] = None,  # TODO (ricmiyam): Remove this and retrieve schema from storage API
+        rebase_source_partition_locator: Optional[PartitionLocator] = None,
+        rebase_source_partition_high_watermark: Optional[int] = None,
```

Comment:
can we get this info directly from `rebase_source_partition_locator` and deltacat storage API?

Reply:
Eventually, I think so - I'll add a TODO for this and open a related issue to propose a storage API for this and leverage it for simplicity of management of catalogs that can support and implement it today, since it will simplify operations in cases where the compactor can determine the old high watermark automatically.

Right now we're lacking any official or consistent specification of how to do this across various data catalogs, but we could consider adding a storage API that simply agrees to return the latest copy-on-write compaction high-watermark for any partition locator, and leave it up to the catalog to determine whether they have a supported implementation for this API.

This currently just defers to the simple answer of "the caller figured out the high watermark somehow" over the more user-friendly ideal of "the compactor will figure this out for you as long as the catalog has implemented an API to automatically retrieve the high watermark."

Reply:
Related Issue: https://github.com/ray-project/deltacat/issues/69